File: app/src/main/java/com/xxf/arch/test/TestFragment.java
package com.xxf.arch.test;

import android.os.Bundle;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.LifecycleOwner;

import com.xxf.arch.XXF;
import com.xxf.arch.presenter.XXFPresenter;

import io.reactivex.rxjava3.core.Observable;


public class TestFragment extends Fragment {
    public TestFragment() {
       // new Presenter(this, null);
    }

//    class Presenter extends XXFPresenter<Object> {
//
//        public Presenter(@NonNull LifecycleOwner lifecycleOwner, Object view) {
//            super(lifecycleOwner, view);
//        }
//
//        @Override
//        public void onCreate() {
//            super.onCreate();
//            Observable.just(1)
//                    .to(XXF.bindLifecycle(getLifecycleOwner()))
//                    .subscribe();
//            Log.d("================>p2", "onCreate");
//        }
//
//        @Override
//        public void onPause() {
//            super.onPause();
//            Log.d("================>p2", "onPause");
//        }
//
//        @Override
//        public void onDestroy() {
//            super.onDestroy();
//            Log.d("================>p2", "onDestroy");
//        }
//    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }
}

File: lib_adapter/lib_adapter_demo/src/main/java/com/xxf/adapter/demo/TestFragment.kt
package com.xxf.adapter.demo

import android.os.Bundle
import android.util.Log
import android.view.View
import androidx.fragment.app.Fragment
import com.xxf.ktx.argumentBinding

class TestFragment : Fragment {
    constructor() : super(R.layout.adapter_test) {
      //  putExtra("xx23");
    }

    private val param by argumentBinding(defaultValue = "默认值啊")
    private val param2:String? by argumentBinding()
    private val param4:String? by argumentBinding ()

    private var param3:String by argumentBinding(defaultValue = "xx")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.d("=======>ex", "" + param);
        param3="xxx"
        Log.d("=======>ex2", "" + param3);
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        param3="xxx4"
        Log.d("=======>ex3", "" + param3);
    }

    override fun onStop() {
        super.onStop()
        param3="xxx5"
        Log.d("=======>ex5", "" + param3);
    }
    override fun onDestroy() {
        super.onDestroy()

    }
}